### DISCUSSION TITLE
Стили

### DISCUSSION THREAD

**Discussion by tipodanet** [k-ivan/visual-bookmarks-chrome#168] [Show and tell]
https://github.com/user-attachments/assets/c2da7099-0f0d-440f-af17-5132a7b77a7d

в отзывах прочитатл , что люди спрашивают как цвет поменять, шрифт. Вот этого я не знаю. Меня только анимация интересует:
Может, кому то покажется избыточной, зато все есть, можно что-то поменять изменить, удалить.
```
@keyframes shad-in {
  33% {box-shadow: 0 0 40px red }
     to {box-shadow: 0 0 10px red }  }

@keyframes shad-out {
  33% {box-shadow: 0 0 40px #8676E3 }
     to {box-shadow: 0 0 0px #8676E3 }  }

#bookmarks.grid {
  transform-style: preserve-3d;
  perspective: 1000px; }

[data-id] {
  --bezier: cubic-bezier(.5, -.5, .5, 1.5);
  animation: shad-out 2s forwards;
  transition: transform 4s var(--bezier) !important;

 &:hover, &:focus {
  animation: shad-in 1s forwards;
  transform: translateZ(50px) rotateX(5deg) rotateY(350deg) rotateZ(5deg) scale(1.5);
  transition: transform 2s var(--bezier) 0s !important;}  }
```

**Comment by k-ivan** [comment]
Спасибо, что поделелились.

У меня давно уже в планах создать страничку с примерами, как и что можно менять, но пока руки не доходят.

> в отзывах прочитатл , что люди спрашивают как цвет поменять, шрифт. Вот этого я не знаю. Меня только анимация интересует: Может, кому то покажется избыточной, зато все есть, можно что-то поменять изменить, удалить.
> 
> ```
> @keyframes shad-in {
>   33% {box-shadow: 0 0 30px red }
>   50% {box-shadow: 0 0 40px red }
>    to {box-shadow: 0 0 10px red }  }
> 
> @keyframes shad-out {
>   33% {box-shadow: 0 0 40px #8676E3 }
>   50% {box-shadow: 0 0 30px #8676E3 }
>    to {box-shadow: 0 0  0px #8676E3 }  }
> 
> #bookmarks.grid    {
>   perspective: 1000px; }
> 
> [data-id] {
>   --bezier: cubic-bezier(.5, -.5, .5, 1.5);
>   z-index: 1;
>   transform-style: preserve-3d;
>   animation: shad-out 2s forwards;
>   transition: transform 4s var(--bezier) !important;
> 
>  &:hover, &:focus {
>   animation: shad-in 1s forwards;
>   z-index: 50;
>   transform: rotateX(5deg) rotateY(350deg) rotateZ(5deg) scale(1.5);
>   transition: transform 2s var(--bezier) 0s !important;}  }
> ```
Важно уточнить, что в этом примере perspective должен быть с большим значением, если закладок много, иначе они будут сильно искажаться при большом кол-ве закладок. Ну анимации с тенями очень сильно могут влиять на производительность анимации.

**Comment by tipodanet** [comment]
Ivan thank you за коммент, я тут все никак GIF  не нарисую. Шас приклею. Каждый сам решит , что ему надо. 
Вообще github не удобный какой-то. Есть форумы с открытым HTML , например, которые на myBB. Там  все работает прямо в окошке.

**Reply by k-ivan** [reply]
Да, но гитхаб не про форум все-таки

**Comment by tipodanet** [comment]
согласен, здесь можно выкинуть половину. Собрал все возможности. Понятно, что все  rotateX rotateZ уберут.  Может, кому-то захочется  наоборот,  rotateX оставить , rotateY убрать. Или вообще всю transform: 

Эти "Визуальные Закладки" редкий случай, когда нимация имеет полезную функцию . Я сделал 3 картинки в ряд для видео. Если сделать 5 или 7 , то особо не разглядишь. Вот здесь и пригодится  transform: scale(2);

**Reply by k-ivan** [reply]
Я скорее имел ввиду, что у Вас в примере 3 больших закладки, если их будет скажем 7 в ряд и список будет большой то perspective:300 исказит их сильно.
Это было небольшое дополнение для тех, кто может столкнуться просто скопировав код.

**Reply by tipodanet** [reply]
для любителей перспективы : perspective: 1000px; -- это для общей сцены.  Есть еще свойство transform: perspective(300px) , влияет только на сам объект. Их действие складывается. Поэкспериментируйте.

**Comment by tipodanet** [comment]
согласен, это первое впечатление. Потом немного заколебает, но я уже привык. Я в последней строке оставил ПАУЗУ  0s .   1  или 2 или 3s  остановит это сумасшествие . И можно будет пользоваться

**Comment by tipodanet** [comment]
Иван, а вот и вопросы подтянулись. В крайних точках этой анимации происходит какой-то странный blur . Вы не знаете причину ?
Blur не сохраняется, ложная тревога. Чуть влияет цвет по контуру.

**Reply by k-ivan** [reply]
О, там может быть много причин. Лень описывать )
Если коротко, то могут влиять: 

- субпиксельный рендеринг
- антиалиасинг
- дробные значения в трансформациях
- аппаратное ускорение
- масштабирование растровых изображений(в случае с миниатюрами)

в данном случае возможно поможет полный перевод всех трансформаций в 3d(хотя может и в обратную сторону сыграть, 4 пункт).

**Reply by tipodanet** [reply]
согласен, Тем более , у меня старый монитор  c TN-матрицей . У кого 4К , таких проблем точно не будет.

**Reply by k-ivan** [reply]
> У кого 4К , таких проблем точно не будет

не факт, очень вероятно, что тоже будут мылиться, тут не от монитора зависит, а скорее от браузера и реализации

**Comment by tipodanet** [comment]
Иван  вот Вам еще идея для [воплощения](https://seregasmyfavoritesites.on.drv.tw/Site/Gallery/Polaroid/).  Скрипт из открытых источников, моя только доп. анимация. Предлагаю Вам повторить его полностью. В разделе "Speed Dial"   Google в Маркете сразу поднимет Вас на 1 место :)

**Reply by k-ivan** [reply]
Нее, это уже не про удобство, это больше про - Вау )

С анимацией важно не перебарщивать )

**Reply by oblominsk** [reply]
Иван, привет. Это я путаюсь, в логинах. Жаль, сдох Polaroid . Скрипт чужой , капризный. Где-то исходники лежат. О часто ломается. Там были картинки с переворотом. Привел ради шутки.

**Comment by tipodanet** [comment]
https://github.com/user-attachments/assets/300cfbfe-239e-427a-9986-9cc8a4d397d2
```
:root {--ugol: 60deg;}

@keyframes bolt {
from { transform: rotateX(0);
       box-shadow: 0 0 0px red; }
 25% { transform: rotateX(calc(-1 * var(--ugol)));
       box-shadow: 0 0 40px red; }
 75% { transform: rotateX(var(--ugol));
       box-shadow: 0 0 40px red; }
  to { transform: rotateX(0);
       box-shadow: 0 0 0px red; }  }

@keyframes shad-out {
  33% {box-shadow: 0 0 40px #8676E3; }
   to {box-shadow: 0 0  0px #8676E3; }  }

#bookmarks.grid {
   transform-style: preserve-3d;
   perspective: 1000px;}

[data-id] {
  transform: perspective(300px);
  transform-origin: center top;
  animation: shad-out 2s forwards;

 &:hover, &:focus {
  animation: bolt 3s linear 0s infinite; }  }
```

**Reply by tipodanet** [reply]
с   animation: сложнее. Она просто сбрасывается , когда hover уходит.  Как сделать ее мягкий возврат в ноль ?

**Reply by k-ivan** [reply]
Попробуйте поиграться со свойством rotate вместо transform
Выставить дефолтно rotate у статичной закладки, и пробуйте заменить все transform на rotate в keyframes. Возможно получится.

**Reply by tipodanet** [reply]
я тоже слышал, что недавно отменили transform: rotate(),  и оставили (дополнили ) просто rotate: . Замечал, что где-то работает, что где-то нет (или опять другая причина не работы).

**Reply by k-ivan** [reply]
caniuse говорит, что работает везде https://caniuse.com/?search=rotate (про основные браузеры)

**Reply by tipodanet** [reply]
еле нашел . только в одном месте про эти короткие свойства: https://doka.guide/css/transform-function/   и на Мозиле https://developer.mozilla.org/en-US/docs/Web/CSS/rotate . Иван, спасибо, хотя это больше offtop  ,но когда то давно я для себя решил, то просто нет поддержки  . Не работает , и черт с ним . если бы вы не показали этот caniuse , я бы не стал искать дальше.
Вот правильный синтаксис   --- `rotate: x 90deg; ` ,  , нет никакого **rotateX:**

**Reply by k-ivan** [reply]
Не за что, если что MDN всегда в помощь

**Reply by tipodanet** [reply]
Если кто будет ставить этот код , то наверное, правильно в последнюю строку  linear вместо ease. Гармонические колебания 💢  требуют замедления по краям. Т.е. время надо пихать внутрь animation:

**Comment by tipodanet** [comment]
Иван, скажите по-простому, можно отключить окно создания миниатюр ? Похожая вещь есть у Вивальди . судя по тому, что у них весь браузер — это нахлобучка над Хромиум 🤣, очевидно, что они сами эти миниатюры они не делали. Значит, у вас общие источники. ♨️ У них миниатюры создаются в фоне. И нет такой возможности включения этого эффектного процесса создания, размером в 100% самого сайта .
В меня 5000 закладок. Боюсь 🙀 нажать кнопку в Вашем расширении ))
Придут к вам оттудова, принесут и 10000 и 15000 закладок. И все равно потребуют отключить это креативное окно. 
........
Но функция задержки -- это  просто BEST OF THE BEST . На Вивальди у меня были черные скрины для сложных сайтов, где даже на 50 Мбит/с страница грузится 5 сек.

**Reply by k-ivan** [reply]
Честно говоря, требуется пояснение, лучше со скриншотам :), так как всей сути не понял

**Reply by tipodanet** [reply]
это окно на каждой миниатюре, вы точно так задумали функционал ? На 5000 закладок 5000 таких окон ? 
(наверное , есть смысл ее удалить, если картинка Дружко не к месту. ) . На  Сереге Дружко главное 
это  не его рожа, а скрипт дождя , который я месяц колупал , чтобы его завести, не мой конечно

**Reply by k-ivan** [reply]
Спасибо!

Раньше была возможность делать захват окна в фоне, с недавних версий политика безопасности обновилась у хрома, теперь можно сделать скрин только на глазах у пользователя, чтобы он видел, что все прозрачно.

Вы можете не переживать, что у Вас много закладок, скриншоты не делаются параллельно, они будут делаться по очереди, не больше одного за раз.

**Reply by tipodanet** [reply]
Вивальди точно использует этот же ресурс. там в фоне, без всяких настроек, как у вас

**Reply by k-ivan** [reply]
Окно всегда одно, как только скрин готов, окно закрывается и открывается для следующего сайта.

**Reply by k-ivan** [reply]
Вивальди наверное генерит скриншоты на своих серверах, мое расширение генерит скрин на вашем компьютере

**Reply by tipodanet** [reply]
> теперь можно сделать скрин только на глазах у пользователя, чтобы он видел, что все прозрачно.

какая связь между скрином и безопасностью ?

**Reply by tipodanet** [reply]
> Вивальди наверное генерит скриншоты на своих серверах, мое расширение генерит скрин на вашем компьютере

так тем более, генерьте сколько влезает в фоне. Но я всегда за галку. Всегда за ВКЛ/ВЫКЛ. У каждого свои отклонения -- кому-то нравится смотреть на 5000 окон, кому-то нет.

**Reply by tipodanet** [reply]
Иван, вы уже выиграли у Вивальди, у него долгие скрипты дают черный экран . Ни Дружко, ни мой Кот там не успевает сгенериться. И Картинки типа Polaroid тоже

**Reply by k-ivan** [reply]
Чтобы сделать скрин, я должен получить от вас разрешение на просмотр и изменение данных на всех сайтах, это очень жёсткое разрешение, если Вы не доверяйте автору, давать такое разрешение не рекомендуется.

Но без этого разрешения, мы не сможем открывать сайт программно и делать скрин.

**Reply by k-ivan** [reply]
> > Вивальди наверное генерит скриншоты на своих серверах, мое расширение генерит скрин на вашем компьютере
> 
> так тем более, генерьте сколько влезает в фоне. Но я всегда за галку. Всегда за ВКЛ/ВЫКЛ. У каждого свои отклонения -- кому-то нравится смотреть на 5000 окон, кому-то нет.

Не, Вы не много не понимайте, я не Вивальди, у меня нет ресурсов, чтобы содержать какие-либо сервера и писать для них бэк часть. Это всё-таки целая компания с большим штатом.

**Reply by tipodanet** [reply]
Какие разрешения ?  Это те, что 3 штуки в настройках Хромиума ? Я всем всегда включаю ОК, включая Инкогнито. Меня это не колышет. Было бы шикарно, если бы все было ВКЛ по умолчанию. Кому надо пусть ОТКЛЮЧАЕТ. Никто не запрещает.  Я уверен, Гугл сразу получит ОК от  99% юзеров. Всем пофиг.

**Reply by tipodanet** [reply]
> , чтобы содержать какие-либо сервера

нет, генерьте на моем компе, но в фоне.

**Reply by k-ivan** [reply]
> Какие разрешения ?  Это те, что 3 штуки в настройках Хромиума ? Я всем всегда включаю ОК, включая Инкогнито. Меня это не колышет. Было бы шикарно, если бы все было ВКЛ по умолчанию. Кому надо пусть ОТКЛЮЧАЕТ. Никто не запрещает.  Я уверен, Гугл сразу получит ОК от  99% юзеров. Всем пофиг.

Да, получит, не спорю. Но сути это не меняет, это разрешение очень опасное. Тем более, что люди могут пользоваться онлайн-банкингом и прочим.

**Reply by k-ivan** [reply]
> > , чтобы содержать какие-либо сервера
> 
> нет, генерьте на моем компе, но в фоне.

Дак я же Вам говорю, что не будет работать в фоне, браузер запрещает это )

**Reply by tipodanet** [reply]
> онлайн-банкингом и прочим.

вот ! Запишите это  в предупреждение "онлайн-банкингом и прочим". , и ГАЛКУ "ВЫКЛ Визуальную генерацию (через включение нужных разрешенией , и картинки где в  Хромиуме искать  какие галки )  "

**Reply by tipodanet** [reply]
А что делать ?* Я не могу пользоваться расширением из-за этого. Если на нажму генерацию , у меня стопорится работа.  Дело  не  в компе, 8 ядер не напрягается, но они постоянно выскакивают . Куда их прятать ? У меня,  конечно,  2 монитора, я мог их прятать на вЫключенный монитор, но не у всех такие возможности.

**Reply by k-ivan** [reply]
> > онлайн-банкингом и прочим.
> 
> вот ! Запишите это  в предупреждение "онлайн-банкингом и прочим". , и ГАЛКУ "ВЫКЛ Визуальную генерацию (через включение нужных разрешенией , и картинки где в  Хромиуме искать  какие галки )  "
> 

В последнем обновлении я уже сделал само разрешение опциональным, и выключенным по умолчанию. Но, повторюсь сути это не меняет.

**Reply by tipodanet** [reply]
не может быть такого. Я не говорю про 5000 закладок. но у любого  найдутся  папки с 100-200 закладок. И что делать ? Смотреть на эту дискотеку ?

**Reply by k-ivan** [reply]
>  А что делать ?* Я не могу пользоваться расширением из-за этого. Если на нажму генерацию , у меня стопорится работа.  Дело  не  в компе, 8 ядер не напрягается, но они постоянно выскакивают . Куда их прятать ? У меня,  конечно,  2 монитора, я мог их прятать на вЫключенный монитор, но не у всех такие возможности. 

Либо пользоваться как есть и смириться, либо не пользоваться. Других вариантов сейчас нет.

**Reply by tipodanet** [reply]
сделайте этому окну `display:none` и пусть оно думает, что  оно есть.

**Reply by k-ivan** [reply]
> не может быть такого. Я не говорю про 5000 закладок. но у любого  найдутся  папки с 100-200 закладок. И что делать ? Смотреть на эту дискотеку ? 

Включить генерацию и пойти пить кофе:)

**Reply by k-ivan** [reply]
> сделайте этому окну `display:none` и пусть оно думает, что  оно есть.

Это не возможно. А если его сдвинуть программно за пределы экрана, браузер сразу меня пошлет.

**Reply by tipodanet** [reply]
Вы меня убиваете ? Иван, вы можете как хозяин Дискуссии  удалить Дружко ? Удаляйте
Я пока новую анимацию приклею

**Reply by k-ivan** [reply]
Удалил )

**Reply by tipodanet** [reply]
Кстати ,он единоборствами занимался, сальто легко делает. Хорошо , что он не видит вашего поста

**Reply by k-ivan** [reply]
> Кстати ,он единоборствами занимался, сальто легко делает. Хорошо , что он не видит вашего поста

Сильное заявление, проверять его не будем)

**Comment by tipodanet** [comment]
https://github.com/user-attachments/assets/4ee8093e-0ff3-42a6-ad04-69a7aa3e327e

```
#bookmarks.grid {
   transform-style: preserve-3d;
   perspective: 1000px;}

[data-id] {
   box-shadow: 0 0 0 red;
   transition: 2s cubic-bezier(0.5, -2, 0.5, 4) !important;

 &:hover, &:focus {
   box-shadow: 0 0 40px red;
   transform: translateZ(100px);
   transition: 1s !important ; }  }
```

**Reply by tipodanet** [reply]
1. легкие глюки -- это глюки рекордера. В реале их нет . По расширению вопросов нет .
2. Самые простые решения  -- самые эффектные.
3. Именно по-этому надо все немного усложнить.

**Comment by tipodanet** [comment]
Иван, а как туда немного JS запихать ?

**Reply by k-ivan** [reply]
Я не делал такую возможность.

**Reply by tipodanet** [reply]
Иван, Вы же понимаете, что сложные анимации, при неизвестном числе  объектов , только через JS . Ну это если хочется всем панелькам  одинаковую анимацию , но с задержкой .  Прикиньте,  задержка при загрузке  0.2s  -- это эффект волны !!!! Это шикарно. И волна из любой простой анимации делает просто WAW !!!!  И тут без JS никак . И тут без 
`elements.forEach((element, index) => {`  ну никак.
Есть 2 варианта : 
1. Вы отрываете полный JS -- конечно , это предпочтительно . Чтобы мы ни в чем не были ограничены. (на Вивальди именно так -- любые свои script.js и styles.css  подключаются)
2. Вы  разрешаете только 
```
element.classList.remove('red'); 
element.classList.add('blue'); 
```
Не знаю, может быть , этого и будет достаточно. 
но все равно, второе без первого не работает. 
Шо вы нам подарите ?

**Reply by k-ivan** [reply]
Я не вижу смысла в этом, по нескольким причинам, это будет нужно паре человек не более и это может потребовать добавление новых разрешений от пользователя, что не приемлемо для того, чтобы пользовались этим пара человек из тысяч.

Я понимаю, что Вам нравятся анимации, но это все-таки не про мое расширение, его суть не в анимациях. Поэтому я не буду Вас обнадёживать, желания такого нет(по причинам выше описанным), но возможно, если это не будет требовать доп. разрешений и не сильно меня обременит, добавлю, но повторюсь, это очень маловероятно.

**Reply by tipodanet** [reply]
не согласен. Чтобы добавить анимацию , юзеру не обязательно что-то знать, только Ctrl+C, Ctrl+V. Этим умением обладает 100% юзеров. И мы не знаем, кто из них любит анимацию  , а кто нет. 
Откуда вы знаете, что  я 1 один из 100 000 ? Все 100 000 могут вставить код, и останутся на это расширении. Я не зря выкладываю видео. Юзер смотрит, копирует и получает. Это все. 
Еще год назад , я был таким же нубом. Ни черта не знал, и не мог отличить CSS  от HTML , набор темных буков.
Т.е. по Вашей терминологии, Иван, если человек не знает CSS, значит, ре не интересуется анимацией. 
Но как ? Я все тот же , только без CSS, До CSS я так же выучил AutoCAD с нуля, там тоже есть 3D-анимация. И не хуже, чем в Maya или Blender.
Я о том, что человек делает свой выбор независимо от своих знаний. А здесь у вашего расширения есть шанс получить развитие и известность среди 100% юзеров. Юзер посмотрит, выберет код, даже если он A от B не отличает. Выберет и останется на этом расширении. 
Таких расширений не мало. Чем они отличаются ? Ничем, все почти одинаковые. 

Самое известное -- FVD Speed Dial . Чем оно берет юзеров ? Анимацией , расположение по радиусу. А чем еще ?  Фонами , картинками -- всем тем, что не относится именно к панелям закладок. Ну там есть глючная синхронизация, из-за которой у многих  юзеров послетали многолетние закладки. Вроде,  они  собираются ее делать платной.

На форуме Вивальди не какая-то мелкая тема. Там целый раздел с сотнями , скорее с 1000-ми тем по кастому -- https://forum.vivaldi.net/category/43/customizations-extensions . И это только на англ. На каждом языке есть свои темы с той же задачей. Это бразуер называют самым кастомизируемым из всех, включая FireFox. Все остальные вообще не про настройки интерфейса. Там тоже подключается CSS , JS . Но уже нужна инструкция в 6 шагов. А это сложно. И все знают, браузер, построенный как надстройка над Хромиумом, не может быть быстрее Хрома, по определению. Но не уходят с него, именно из-за его кастомизации. 

Да весь дизайн -- это то,   что не относится к самому предмету, а только к его внешнему виду. Иван, Вы отрицаете само понятие дизайна. 

Я уверен, что многим людям вообще наплевать на сами закладки. Я могу их достать и через Панель Закладок,  и через  столб из Меню. Зачем эти квадраты, которые занимают лишнее место на экране ?

Сами Визуальные Закладки непрактичны -- это ТОЛЬКО про дизайн. А вы Иван, это отрицаете.

**Reply by k-ivan** [reply]
Это Ваше мнение, я останусь при своем.

**Reply by k-ivan** [reply]
И конечно в корне не согласен с утверждением, "кому нужны квадраты, можно и из браузера достать", наверное тем, кто ставит их, и не достает из браузера.

**Reply by tipodanet** [reply]
Иван, вы говорите о каких-то разрешениях. Я знаю CSS и половину JS, но я не знаю, где эти разрешения. И за многолет мой комп не сгорел, от того, что они включены или нет , не в курсе. 
Все решается одним Предупреждением. текстом на английском о последствиях. Все . 
Делайте как Гуугл в DevTools . Вы же знаете , там нельзя копировать текст в Console . Но стоит только прочитать "нельзя ничего вставлять плохого " , что-то там написать типа "я понял", и пожалуйста, вставляй что хочешь. Хоть CSS , хоть JS, хоть С++.
Это явления одного порядка. Почему Гуглу можно,  а вам нельзя ?

**Reply by k-ivan** [reply]
Вы скорее всего не знайте, что такое разрешения для расширений, вокруг них все и крутится, это основа. Из-за них расширение может проходить долгие проверки для допуска, вокруг него уже может быть много отпугивающих уведомлений от браузера, которые могут отпугивать. Я встречал уже много пользователей, которых пугало кол-во разрешений, которые запрашивало расширение, то что Вы с этим не сталкивались(или игнорируйте конфиденциальность и риски быть хакнутым или потерять важные данные) не говорит о том, что все не париться об этом.
Поэтому, моя задача, стараться не запрашивать, то что не нужно для работы(или нужно очень-очень узкому кругу людей). Я говорю исходя из своего опыта, почти никому это будет не нужно.

**Reply by tipodanet** [reply]
Иван , не знаю, какие у кого есть проблемы с безопасностью, если юзер САМ включает доступ к JS , после прочтения предупреждения. По умолчанию доступа же нет. Не понимаю, в чем может быть проблема. Если юзер знает , что такое JS,  я думаю, он понимает, что он копирует и вставляет. Боишься за свои данные, ничего не разрешай. Какие вопросы. 
Это также должны понимать и в Google,  при проверке расширения.

без всяких намеков . Стал искать вдруг у кого открыто доступ к JS . Вижу "Визуальные Онлайн Закладки для Chrome"  -- у них видео, где скрин формируется мгновенно и в фоне. Быстрее, чем у вивальди. Не думаю, что у этого расширения свои сервера (довод с потолка)))). 
Может, есть другой способ создания скринов ?

**Reply by k-ivan** [reply]
Проблемы у пользователя не возникают, он просто проходит мимо моего расширения, видя большой набор разрешений, которые потенциально могут следить, подменять, красть данные.

> Стал искать вдруг у кого открыто доступ к JS . Вижу "Визуальные Онлайн Закладки для Chrome" -- у них видео, где скрин формируется мгновенно и в фоне. Быстрее, чем у вивальди. Не думаю, что у этого расширения свои сервера

Не обижайтесь, но зачем писать, не разобравшись, как оно там работает, это расширение использует свой сервер, оно не делает скрины на вашем компьютере. Вы легко можете сами это проверить, сделав скрин сайта на котором вы авторизованы, скрин с их сервера покажет форму входа, а не Ваш аккаунт. 
вот скрин, где четко видно, то, что идет запрос на их сервер, которые в ответ присылает данные с картинкой
![image](https://github.com/user-attachments/assets/7d0838c3-2f41-401d-b157-00c725f24375)

Я знаю, что говорю, если генерировать скрины локально, с недавних пор нельзя прятать окно с сайтом за пределы экрана. Все Blink браузеры будут ругаться и не дадут сделать захват экрана.

**Comment by tipodanet** [comment]
https://github.com/user-attachments/assets/c769381b-3102-48f4-8885-e95d1f95dbcb
```
@keyframes scale-up {
  50% { transform-origin: center ;
	transform: rotateX(10deg) translate3D(0, 15px, 15px);
	box-shadow: 0 0 20px red;}
   to { transform: rotateX(0) translate3D(0, 30px, 30px);
	box-shadow: 0 0 40px red;}  }

@keyframes scale-down {
 from { transform: rotateX(0) translate3D(0, 30px, 30px);
	box-shadow: 0 0 40px red;}
  50% { transform-origin: center ;
	transform: rotateX(10deg) translate3D(0, 15px, 15px);
	box-shadow: 0 0 20px red;}
   to { transform: rotateX(0) translate3D(0, 0, 0);
	box-shadow: 0 0 0 red;}  }

#bookmarks.grid {
	transform-style: preserve-3d; 
	perspective: 1000px;}

[data-id] {
	animation: scale-down .8s linear;
  &:hover, &:focus {
        animation: scale-up .3s linear forwards; } }
```

**Reply by tipodanet** [reply]
github не знает про HEVC кодек.

**Reply by tipodanet** [reply]
Иван, эта вечная проблема с z-index; Получается, каждая панель в своем контексте.  z-index:100 не работает;  Подъем по Z тоже не работает. Как поднять hover-панель над всеми ?

**Reply by k-ivan** [reply]
Попробуйте 
у `scale-down`  в начале `z-index: 1` на 50% `z-index: 0`
у `scale-up` 50% `z-index: 0` и 100% `z-index: 1`
если поиграться с z-index внутри keyframes, возможно поможет

Я бы рекомендовал для подобных интерактивных с мышью вещей использовать `transition`,  это гораздо удобнее. Вам не нужно будет делать обратный шаг, + анимация не будет дерганой при резких `hover` мышкой.

PS: пожалуйста форматируйте код, прежде чем вставить его, в таком виде он выглядит трудночитаемым

**Reply by tipodanet** [reply]
у Notepad и githab разное отношение к TAB.
никакие z-index:  не помогают. Активная вкладка всегда  ниже соседней.
transition не может заменить animation  на сложных движениях. А animation не может догнать transition при плавных возвратах.
придется убирать rotateX()

**Reply by tipodanet** [reply]
все, победа. Z-index -- зло,  бесполезная хрень. Все можно сделать через **transform-style: preserve-3d;  и  translateZ()**

**Reply by tipodanet** [reply]
жаль отказываться от animation  . Только она может собрать все движения  в кучку. Можно только порекомендовать сократить время :hover до  0.3 - 0.2s . Чтобы hover успевал доезжать до конца. А вот обратную анимацию можно делать максимально длинной и плавной.

**Reply by k-ivan** [reply]
> у Notepad и githab разное отношение к TAB.

тут дело не только в табах, тут еще и переносы отсутствуют в ряде мест, что очень сильно затрудняет чтение

**Reply by oblominsk** [reply]
я не обратил внимания, сразу, : я просто указал  стиль для тренировки , и минимального кода. 
```
[data-id] {
	animation: scale-down .8s linear;
```
А получился классный эффект начального движения всей страницы. 
Если делать начальное положение статичное, то это еще пару строк кода. 
Скорее всего , логика извращенная. Если переписать @keyframes  с 0px до 30px  , то будет правильно и скучно

**Comment by BigStacy** [comment]
Здравствуйте, Иван. Понравилось Ваше расширение. Можно ли изменить размер иконок как на втором изображении? На первом скриншоте показал как это выглядит сейчас, кажется они стали меньше.
![Screenshot 2024-09-21 at 02-18-03 Визуальные закладки](https://github.com/user-attachments/assets/9b590d29-a034-469d-96c9-4db89620925f)
![303437](https://github.com/user-attachments/assets/356dd093-6291-4f30-8817-4493b8029cea)

**Reply by k-ivan** [reply]
Здравствуйте! Рад, что Вам нравится)

На данный момент есть 3 варианта.
1) по умолчанию, отображается фавиконки размером 32px(увеличить можно, но нет смысла, так как они станут мыльные)
2) в настройках включить опцию, **Внешние логотипы от Clearbit**(это сервис предоставляющий логотипы) У них размер будет как на втором скриншоте. Но это сторонний сервис, поэтому у него может не быть каких-то логотипов, он может быть не доступен(то есть его работу гарантировать не могу)
3) загрузка изображений с компьютера(также как и во втором варианте можно добиться размеров как на скрине.

P.S. Буквально, пару версий назад вместо clearbit сервиса, я пробовал использовать logo.dev, мне понравились их лого, но вынужден был отключить, они попросили перейти на платный тариф из-за кол-ва запросов. Но можно добавить опцию вставки своего сервиса, чтобы иметь альтернативу Clearbit, пока думаю надо ли оно.

**Reply by k-ivan** [reply]
На скриншоте, использовались 2 или 3 варианты.

**Reply by BigStacy** [reply]
Спасибо, что ответили так быстро

**Reply by k-ivan** [reply]
Рад помочь, если есть вопросы по поводу кастомизации внешнего вида, всегда рад.

Давно в планах сделать страничку с подборкой решений, как изменить внешний вид по себя.

**Reply by KomAuras** [reply]
> 2\. Внешние логотипы от Clearbit
Кстати сервис для России не работает... Amazon CloudFront блокирует его работу.
Предыдущий работал.

**Reply by KomAuras** [reply]
> logo.dev, мне понравились их лого, но вынужден был отключить, они попросили перейти на платный тариф из-за кол-ва запросов
Можно реализовать возможность сохранения и вставки своего токена в их ссылку. Бесплатного плана на 500000 вызовов API должно людям хватить...
Или как выпишете возможность вызова своей ссылки с макросом типа
https://img.logo.dev/@SiteName?token=XXXX

**Reply by k-ivan** [reply]
Наверное вернусь к ссылке в настройках, самый оптимальный вариант в текущей ситуации.

**Reply by KomAuras** [reply]
> Наверное вернусь к ссылке в настройках, самый оптимальный вариант в текущей ситуации.

Работает. Спасибо большое!

**Reply by BigStacy** [reply]
> > Наверное вернусь к ссылке в настройках, самый оптимальный вариант в текущей ситуации.
> 
> Работает. Спасибо большое!

Как настроили? Я пытался вводить разные сайты, но логотипы не подхватывает. Если загружать свои, то в ячейках любого размера логотип растягивается...

**Reply by k-ivan** [reply]
@BigStacy какие именно сайты вы пытались использовать?

На данный момент я знаю только 2(logo.dev и clearbit), с ними работает.

**Reply by BigStacy** [reply]
@k-ivan 
> @BigStacy какие именно сайты вы пытались использовать?
> 
> На данный момент я знаю только 2(logo.dev и clearbit), с ними работает.

А, понял, думал можно поставить ссылку на любой сайт где есть иконки. С logo.dev не разобрался в итоге. Хотел скачивать оттуда и загружать вручную, но иконки любого размера, автоматически растягиваются и залипают по вертикали к краям плитки, по горизонтали всё нормально.

**Reply by k-ivan** [reply]
> @k-ivan
> 
> > @BigStacy какие именно сайты вы пытались использовать?
> > На данный момент я знаю только 2(logo.dev и clearbit), с ними работает.
> 
> А, понял, думал можно поставить ссылку на любой сайт где есть иконки. С logo.dev не разобрался в итоге. Хотел скачивать оттуда и загружать вручную, но иконки любого размера, автоматически растягиваются и залипают по вертикали к краям плитки, по горизонтали всё нормально.

Не, нужны сервисы, которые специально создавались для таких нужд.

> автоматически растягиваются и залипают по вертикали к краям плитки

Так и задумано, чтобы в случае очень крупной картинки, весь ее контекст был виден. Это можно изменить, если Вы хотите, чтобы картинка заполняла всю плитку, то можно переопределить стиль по умолчанию, прописав собственный стиль
```css
.bookmark__img--contain {
  background-size: cover;
}
```
![image](https://github.com/user-attachments/assets/7d1b7f4a-df95-42cb-8a35-5e2c90334777)

Или, необходимо подбирать картинку, которая по размеру будет равна или чуть больше плитки.

> С logo.dev не разобрался в итоге

Там все просто, Вам необходимо пройти регистрацию, чтобы получить токен ключ.
После регистрации, в личном кабинете Вы добавите свой токен и сможете вставить его в урл. Подробнее https://docs.logo.dev/introduction#getting-an-api-key
![image](https://github.com/user-attachments/assets/87a3f029-de88-4444-a4bf-8cb31547b5ce)


После этого, вставьте урл с токеном в настройки закладок
```
https://img.logo.dev/{{website}}?token=скопируйте_сюда_токен
```
![image](https://github.com/user-attachments/assets/6818dd0c-85b0-4df7-95b5-97cf7ff029ac)

**Reply by BigStacy** [reply]
@k-ivan 
Благодарю. Всё получилось. Некоторых логотипов не хватает, лично мне. А есть ли команда в настройках CSS, чтобы не растягивала иконку по вертикали, чтобы она не прилипала к низу и верху? Я стараюсь небольшие иконки ставить, 60х60 примерно, поэтому не хотелось бы растягивания. Спасибо, что помогаете и разъясняете

**Reply by k-ivan** [reply]
> @k-ivan Благодарю. Всё получилось. Некоторых логотипов не хватает, лично мне. А есть ли команда в настройках CSS, чтобы не растягивала иконку по вертикали, чтобы она не прилипала к низу и верху? Я стараюсь небольшие иконки ставить, 60х60 примерно, поэтому не хотелось бы растягивания. Спасибо, что помогаете и разъясняете

Рад помочь!

Если картинка заметно меньше самой плитки, в таком случае, можно использовать
```css
.bookmark__img--contain {
    background-size: auto;
}
```
или вместо `auto` Вы можете задать свое значение в px
например
```css
.bookmark__img--contain {
    background-size: 50px;
}
```

**Reply by BigStacy** [reply]
@k-ivan, большое спасибо, получилось как хотел
